flink-runtime/src/main/java/org/apache/flink/runtime/state/StateBackendLoader.java

```diff
@@ -47,6 +52,13 @@
     //  Configuration shortcut names
     // ------------------------------------------------------------------------
 
+    private static final Set<String> STATE_BACKEND_CAN_BE_PROXIED =
+            new HashSet<>(
+                    Arrays.asList(
+                            "org.apache.flink.contrib.streaming.state.RocksDBStateBackend",
```

Comment:
The reason is because the RocksDBStateBackend is in a different module (dependency problem), you can not use "RocksDBStateBackend.class" directly.

the loader is in runtime
rocksdb module depends on runtime.

Reply:
You can check how rocks db is loaded, it is using the class name as well.

Reply:
But I think you make a good point here, we should not hardcoded the can-be-proxied state backends.

Reply:
But what is the reason to limit proxying to these classes in the first place? Why not proxy everything?
Or, if we want to avoid double-proxying then maybe check that specific class?